base/applications/rapps/gui.cpp

```diff
@@ -1332,296 +1371,717 @@ class CAppsListView :
+        }
+        TableViewMode = Mode;
+        m_AppsInfo->SetWelcomeText();
+
+        HMENU lvwMenu = ::GetMenu(m_ListView->m_hWnd);
+        switch (Mode)
+        {
+        case TableViewEmpty:
+        default:
+            EnableMenuItem(lvwMenu, ID_REGREMOVE, MF_GRAYED);
+            EnableMenuItem(lvwMenu, ID_INSTALL, MF_GRAYED);
+            EnableMenuItem(lvwMenu, ID_UNINSTALL, MF_GRAYED);
+            EnableMenuItem(lvwMenu, ID_MODIFY, MF_GRAYED);
+            break;
+
+        case TableViewInstalledApps:
+            EnableMenuItem(lvwMenu, ID_REGREMOVE, MF_ENABLED);
+            EnableMenuItem(lvwMenu, ID_INSTALL, MF_GRAYED);
+            EnableMenuItem(lvwMenu, ID_UNINSTALL, MF_ENABLED);
+            EnableMenuItem(lvwMenu, ID_MODIFY, MF_ENABLED);
+            break;
+
+        case TableViewAvailableApps:
+            EnableMenuItem(lvwMenu, ID_REGREMOVE, MF_GRAYED);
+            EnableMenuItem(lvwMenu, ID_INSTALL, MF_ENABLED);
+            EnableMenuItem(lvwMenu, ID_UNINSTALL, MF_GRAYED);
+            EnableMenuItem(lvwMenu, ID_MODIFY, MF_GRAYED);
+            break;
+        }
+        return TRUE;
+    }
+
+    BOOL AddInstalledApplication(CInstalledApplicationInfo * InstAppInfo, LPVOID param)
+    {
+        if (TableViewMode != TableViewInstalledApps)
+        {
+            return FALSE;
+        }
+        return m_ListView->AddInstalledApplication(InstAppInfo, param);
+    }
+
+    BOOL AddAvailableApplication(CAvailableApplicationInfo * AvlbAppInfo, BOOL InitCheckState, LPVOID param)
+    {
+        if (TableViewMode != TableViewAvailableApps)
+        {
+            return FALSE;
+        }
+        return m_ListView->AddAvailableApplication(AvlbAppInfo, InitCheckState, param);
+    }
+
+    void CheckAll()
+    {
+        m_ListView->CheckAll();
+        return;
+    }
+
+    PVOID GetFocusedItemData()
+    {
+        return m_ListView->GetFocusedItemData();
+    }
+
+    int GetItemCount()
+    {
+        return m_ListView->GetItemCount();
+    }
+
+    // this function is called when a item of listview get focus.
+    // CallbackParam is the param passed to listview when adding the item (the one getting focus now).
+    BOOL ItemGetFocus(LPVOID CallbackParam)
+    {
+        switch (TableViewMode)
+        {
+        case TableViewInstalledApps:
+            return m_AppsInfo->ShowInstalledAppInfo((CInstalledApplicationInfo *)CallbackParam);
+
+        case TableViewAvailableApps:
+            return m_AppsInfo->ShowAvailableAppInfo((CAvailableApplicationInfo *)CallbackParam);
+
+        case TableViewEmpty:
+        default:
+            m_AppsInfo->SetWelcomeText();
+            return FALSE;
+        }
+    }
+
+    // this function is called when a item of listview is checked/unchecked
+    // CallbackParam is the param passed to listview when adding the item (the one getting focus now).
+    BOOL ItemCheckStateChanged(BOOL bChecked, LPVOID CallbackParam);
+    
+};
+
+class CSideTreeView :
```

Comment:
It might make sense to move a few of those to their own cpp file, instead of dropping everything in gui.cpp

Reply:
of course, having their own cpp file is better but this means they need their own .h file too (some classes here are referencing each other). It's not a small work and this PR is already too big IMO🤣
I would like to do this, probably next PR but not this🤣